logger/drain/factory_test.go

```diff
@@ -0,0 +1,66 @@
+package drain
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestGetUsingInvalidValues(t *testing.T) {
+	drainStrs := []string{"bogus"}
+	for _, drainStr := range drainStrs {
+		_, err := NewDrain(drainStr)
+		if err == nil ||
```

Comment:
Can you please remove the newline after `||` here. It is admittedly visible without scrolling on Github but it took my eyes quite some time to adjust to it. See https://github.com/golang/go/wiki/CodeReviewComments#line-length